src/stream/src/executor/source/source_executor.rs

```diff
@@ -86,10 +93,16 @@ impl<S: StateStore> SourceExecutor<S> {
         streaming_metrics: Arc<StreamingMetrics>,
         expected_barrier_latency_ms: u64,
     ) -> Result<Self> {
+        // Using vnode range start for row id generator.
+        let vnode_id = vnodes.next_set_bit(0).unwrap_or(0);
         Ok(Self {
             actor_id,
             source_id,
             source_desc,
+            row_id_generator: RowIdGenerator::with_epoch(
+                vnode_id as u32,
```

Comment:
Shall we rename this field from `worker_id` to `vnode_id`?

https://github.com/singularity-data/risingwave/blob/60234a4302f4163c70dac312764fb244913b6b17/src/source/src/row_id.rs#L22-L38